services/workflows-service/src/workflow/workflow.service.ts

```diff
@@ -716,6 +717,13 @@ export class WorkflowService {
         },
       });
     } else {
+      if (manualReviewWorkflow.state === 'revision') {
```

Comment:
Check if the "headless example" machine is in final state

Reply:
This logic is in `handleRuntimeFinalState` which runs when the headless example machine reaches its final state.